src/util/config.h

```diff
@@ -140,6 +146,9 @@ class configt
   static std::string this_operating_system();
 
   static triple host();
+
+  // For caching
+  std::set<assert_pair> caching_db;
```

Comment:
I think you want std::unordered_set as it is hash-based, std::set relies on the order.